### ISSUE TITLE
Lint non-code syntax for variables in documentation comments

### ISSUE THREAD

**Issue by gnzlbg** [rust-lang-nursery/rust-clippy#2178]
Here:

```rust
/// This function adds a with `b`.
fn add(a: u32, b: u32) -> u32 { a + b };
```

note how in the documentation comment `'b'` is quoted but `a` is not. We should lint about these.

**Comment by oli-obk** [comment]
Your example is the one that's the most impossible... `a` is an english word.

We'd be getting false positives left and right even if we'd have some english language parser. Not really sure if we can act on this. Maybe somebody's got ideas how to do it.

**Comment by gnzlbg** [comment]
> Your example is the one that's the most impossible... a is an english word.

Indeed. 

How do Rust IDEs infer what variables do? 

For example in C++ there is the Doxygen format which IDEs use it to give detailed descriptions of the arguments and return type, something like:

```rust
/// Adds two values. 
///
/// # Arguments
///
/// - `T`: type of the arguments to add
/// - `a`: argument to add
/// - `b`: argument to add
///
/// # Returns
///
/// Sum of `a` and `b`.
fn add<T>(a: T, b: T) -> T;
```

**Comment by oli-obk** [comment]
I'd really want to have

```rust
fn add<T>(
    /// super first argument
    a: T,
    /// horrible second argument
    b: T,
/// Returns sum of `a` and `b`
) -> T;
```

Maybe I'll do an rfc next year

**Comment by gnzlbg** [comment]
@oli-obk we should ping the new `rustdoc` repo on this as well. It might be easier to prototype this there than in the old rustdoc.

**Comment by killercup** [comment]
@gnzlbg cf https://github.com/rust-lang-nursery/rust-clippy/issues/1007#issuecomment-319705879 re: doc format
